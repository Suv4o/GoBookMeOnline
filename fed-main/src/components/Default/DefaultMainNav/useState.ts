import { reactive, toRefs } from 'vue'

const state = reactive({
  isSignInButtonShown: true,
  isSignUpButtonShown: true,
})

export default () => {
  function setSignInButtonShown(isShown = true) {
    state.isSignInButtonShown = isShown
  }

  function setSignUpButtonShown(isShown = true) {
    state.isSignUpButtonShown = isShown
  }

  return {
    ...toRefs(state),
    setSignInButtonShown,
    setSignUpButtonShown,
  }
}
