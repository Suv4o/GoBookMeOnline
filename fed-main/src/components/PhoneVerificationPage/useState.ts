import { reactive, toRefs } from 'vue'

const state = reactive({
  fullName: '',
  phoneNumber: '',
})

export default () => {
  function setFullName(name = '') {
    state.fullName = name
  }

  function setPhoneNumber(phone = '') {
    state.phoneNumber = phone
  }

  return {
    ...toRefs(state),
    setFullName,
    setPhoneNumber,
  }
}
