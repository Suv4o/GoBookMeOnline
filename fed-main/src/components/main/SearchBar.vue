<script lang="ts">
export default {
  name: 'SearchBar',
}
</script>

<script setup lang="ts">
import SearchInput from './SearchInput.vue'
import LocationInput from './LocationInput.vue'
</script>

<template>
  <div class="flex flex-row mx-auto max-w-3xl">
    <div class="basis-2/3 mr-4">
      <search-input />
    </div>
    <div class="basis-1/3"><location-input /></div>
  </div>
</template>
