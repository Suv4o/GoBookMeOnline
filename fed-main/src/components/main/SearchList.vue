<script lang="ts">
export default {
  name: 'SearchList',
}
</script>

<script setup lang="ts">
import BookSearchList from '../global/BookSearchList/BookSearchList.vue'
import BookPagination from '../global/BookPagination/BookPagination.vue'
</script>

<template>
  <book-search-list title="Locations" type="location" />
  <book-search-list title="People" type="person" />
  <div class="mt-4">
    <book-pagination />
  </div>
</template>
