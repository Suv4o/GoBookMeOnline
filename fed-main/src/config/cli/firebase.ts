#!/usr/bin/env node
import * as yargs from 'yargs'
import * as dotenv from 'dotenv'
import * as admin from 'firebase-admin'
const env = dotenv.config({ path: `./env/.env.${process.env.NODE_ENV}` })

const userIds = [] as string[]

const app: admin.app.App = admin.initializeApp({
  credential: admin.credential.cert({
    type: env.parsed?.VITE_FIREBASE_TYPE,
    project_id: env.parsed?.VITE_FIREBASE_PROJECT_ID,
    private_key_id: env.parsed?.VITE_FIREBASE_PRIVATE_KEY_ID,
    private_key: Buffer.from(env.parsed?.VITE_FIREBASE_PRIVATE_KEY || '', 'base64').toString(),
    client_email: env.parsed?.VITE_FIREBASE_CLIENT_EMAIL,
    client_id: env.parsed?.VITE_FIREBASE_CLIENT_ID,
    auth_uri: env.parsed?.VITE_FIREBASE_AUTH_URI,
    token_uri: env.parsed?.VITE_FIREBASE_TOKEN_URI,
    auth_provider_x509_cert_url: env.parsed?.VITE_FIREBASE_AUTH_PROVIDER_X509_CERT_URL,
    client_x509_cert_url: env.parsed?.VITE_FIREBASE_CLIENT_X509_CERT_URL,
  } as any),
})

const getAllUsersIds = async (nextPageToken?: string) => {
  try {
    const listUsersResult = await app.auth().listUsers(1000, nextPageToken)
    listUsersResult.users.forEach(userRecord => {
      userIds.push(userRecord.uid)
    })
    if (listUsersResult.pageToken) {
      getAllUsersIds(listUsersResult.pageToken)
    }
  } catch (error) {
    console.log('Error listing users:', error)
  }
}

yargs.command(
  'drop',
  'firebase drop users',
  async () => {
    await getAllUsersIds()
    if (userIds.length) {
      try {
        const deleteUsersResult = await app.auth().deleteUsers(userIds)
        console.log(`Successfully deleted ${deleteUsersResult.successCount} users`)
        console.log(`Failed to delete ${deleteUsersResult.failureCount} users`)
      } catch (error) {
        console.log('Error deleting users:', error)
      }
    } else {
      console.log('No users to delete')
    }
  },
  argv => {
    console.log(argv)
  }
).argv
