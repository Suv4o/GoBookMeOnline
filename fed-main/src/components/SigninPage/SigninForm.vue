<script lang="ts">
export default {
  name: 'SigninForm',
}
</script>

<script setup lang="ts">
import { Auth } from '@firebase/auth'
import { signInWithPopup, GoogleAuthProvider } from 'firebase/auth'
import { inject, ref, reactive } from 'vue'
import router from '../../router'
import { useAuthStore } from '../../store/auth'
import { NotificationTypes } from '../../store/notification'
import { Roles } from '../../types/enums'
import { Assertions } from '../../types/guards'
import { useFetch } from '../../utils/composables/fetch'
import { useNotification } from '../../utils/composables/notification'
import { ResponseValidator, useValidator } from '../../utils/composables/validator'
import { parseErrorMessage, parseFirebaseError } from '../../utils/helpers'
import useState from '../PhoneVerificationPage/useState'
import { CurrentUserDetails } from '../SignupPage/SignupForm.vue'

const $auth = inject('$auth') as Auth
const useAuthState = useAuthStore()
const googleProvider = new GoogleAuthProvider()

const phoneOrEmail = ref('')
const isProcessing = ref(false)

const isValid = reactive({
  phoneOrEmail: { valid: true, message: '' },
} as Pick<ResponseValidator, 'phoneOrEmail'>)

async function signInWithEmail() {
  try {
    const { error } = await useFetch({
      url: '/user/signin-email',
      method: 'POST',
      body: {
        email: phoneOrEmail.value,
      },
      credentials: false,
    })

    if (error.value) {
      throw new Error(parseErrorMessage(error.value.message))
    }
  } catch (error) {
    Assertions.isError(error)
    throw new Error(error.message)
  }
}

async function signInWithPhone() {
  try {
    const { error, data } = await useFetch({
      url: '/user/signin-phone',
      method: 'POST',
      body: {
        phoneNumber: phoneOrEmail.value,
      },
      credentials: false,
    })

    if (error.value) {
      throw new Error(parseErrorMessage(error.value.message))
    }

    const user = data.value as CurrentUserDetails | null

    if (!user) {
      throw new Error('This user cannot be created!')
    }

    return user
  } catch (error) {
    Assertions.isError(error)
    throw new Error(error.message)
  }
}

async function signUpWithGoogle() {
  try {
    await signInWithPopup($auth, googleProvider)
    await storeUserToDatabase()
    router.push({ name: 'home', query: { 'successfully-signed': 'true' } })
  } catch (error) {
    Assertions.isError(error)
    clearInputs()
    const readableError = parseFirebaseError(error.message)
    if (readableError) {
      useNotification({ type: NotificationTypes.Error, title: 'Error', message: readableError })
    } else {
      useNotification({ type: NotificationTypes.Error, title: error.name, message: error.message })
    }
  }
}

async function storeUserToDatabase() {
  try {
    const { error, data } = await useFetch({
      url: '/user/signup-with-provider',
      method: 'POST',
      body: {
        role: Roles.USER_DEFAULT,
      },
    })

    if (error.value) {
      throw new Error(parseErrorMessage(error.value.message))
    }

    const user = data.value as CurrentUserDetails | null

    if (!user) {
      throw new Error('This user cannot be created!')
    }

    useAuthState.user = {
      uid: user.uid,
      firstName: user.firstName,
      lastName: user.lastName,
      role: user.role,
      email: user.email,
      phoneNumber: user.phoneNumber,
      displayName: user.displayName,
      photoURL: user.photoURL,
      emailVerified: user.emailVerified,
    }
  } catch (error) {
    Assertions.isError(error)
    throw new Error(error.message)
  }
}

function clearInputs() {
  phoneOrEmail.value = ''
}

async function signInUser(event: Event) {
  event.preventDefault()

  const { validProps } = useValidator({
    phoneOrEmail: phoneOrEmail.value,
  })

  isValid.phoneOrEmail = validProps.phoneOrEmail

  if (!validProps.phoneOrEmail.valid) {
    phoneOrEmail.value = ''
  }

  if (validProps.phoneOrEmail.valid) {
    if (!validProps.phoneOrEmail.isMobile) {
      try {
        isProcessing.value = true
        useAuthState.isUserAuthCompleted = false
        await signInWithEmail()
        clearInputs()
        useNotification({
          type: NotificationTypes.Success,
          title: 'Email has been sent',
          message: 'Please check your email and click the link to sign in.',
        })
        isProcessing.value = false
        useAuthState.isUserAuthCompleted = true
      } catch (error) {
        isProcessing.value = false
        useAuthState.isUserAuthCompleted = true
        Assertions.isError(error)
        const readableError = parseFirebaseError(error.message)
        if (readableError) {
          useNotification({ type: NotificationTypes.Error, title: 'Error', message: readableError })
        } else {
          useNotification({ type: NotificationTypes.Error, title: error.name, message: error.message })
        }
      }
    } else {
      try {
        isProcessing.value = true
        useAuthState.isUserAuthCompleted = false
        const { displayName, phoneNumber } = await signInWithPhone()
        const { setFullName, setPhoneNumber } = useState()
        setFullName(displayName)
        setPhoneNumber(phoneNumber)
        clearInputs()
        router.push({ name: 'phone-verification' })
        isProcessing.value = false
        useAuthState.isUserAuthCompleted = true
      } catch (error) {
        isProcessing.value = false
        useAuthState.isUserAuthCompleted = true
        Assertions.isError(error)
        clearInputs()
        const readableError = parseFirebaseError(error.message)
        if (readableError) {
          useNotification({ type: NotificationTypes.Error, title: 'Error', message: readableError })
        } else {
          useNotification({ type: NotificationTypes.Error, title: error.name, message: error.message })
        }
      }
    }
  }
}
</script>

<template>
  <div class="mx-auto w-full max-w-sm lg:w-96">
    <h2 class="mt-6 text-3xl font-extrabold text-gray-900">Sign in to your account</h2>

    <div class="mt-8">
      <div>
        <div>
          <p class="text-sm font-medium text-gray-700">Sign in with Google</p>

          <div class="mt-1 grid grid-cols-1 gap-3">
            <div>
              <a
                href="javascript:;"
                data-testid="Sign in with Google"
                class="w-full inline-flex justify-center py-2 px-4 border border-gray-300 rounded-md shadow-sm bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"
                @click="signUpWithGoogle"
              >
                <span class="sr-only">Sign in with Google</span>
                <svg class="w-5 h-5" aria-hidden="true" fill="currentColor" viewBox="0 0 210 210">
                  <path
                    d="M0,105C0,47.103,47.103,0,105,0c23.383,0,45.515,7.523,64.004,21.756l-24.4,31.696C133.172,44.652,119.477,40,105,40
	c-35.841,0-65,29.159-65,65s29.159,65,65,65c28.867,0,53.398-18.913,61.852-45H105V85h105v20c0,57.897-47.103,105-105,105
	S0,162.897,0,105z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>

        <div class="mt-6 relative">
          <div class="absolute inset-0 flex items-center" aria-hidden="true">
            <div class="w-full border-t border-gray-300" />
          </div>
          <div class="relative flex justify-center text-sm">
            <span class="px-2 bg-gray-50 text-gray-500"> Or continue with </span>
          </div>
        </div>
      </div>

      <div class="mt-6">
        <form class="space-y-6" data-testid="Sign in a user" @submit="signInUser">
          <div>
            <label for="mobile-or-email" class="block text-sm font-medium text-gray-700">
              Mobile Number or Email
            </label>
            <div class="mt-1">
              <input
                id="mobile-or-email"
                v-model="phoneOrEmail"
                data-testid="Phone or Email"
                :disabled="isProcessing"
                name="mobile-or-email"
                type="text"
                autocomplete="email|tel"
                required="true"
                class="appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none sm:text-sm"
                :class="`${
                  !isValid.phoneOrEmail.valid
                    ? 'ring-red-700 border-red-700'
                    : 'focus:ring-teal-500 focus:border-teal-500'
                }`"
              />
              <p v-if="!isValid.phoneOrEmail.valid" class="text-sm text-red-700 mt-1">
                {{ isValid.phoneOrEmail.message }}
              </p>
            </div>
          </div>

          <div>
            <button
              type="submit"
              class="w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-white bg-teal-600 hover:bg-teal-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
            >
              <svg
                v-if="isProcessing"
                class="animate-spin -ml-1 mr-3 h-5 w-5 text-white"
                xmlns="http://www.w3.org/2000/svg"
                fill="none"
                viewBox="0 0 24 24"
              >
                <circle class="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" stroke-width="4"></circle>
                <path
                  class="opacity-75"
                  fill="currentColor"
                  d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
                ></path>
              </svg>
              {{ isProcessing ? 'Signing in...' : 'Sign in' }}
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</template>
