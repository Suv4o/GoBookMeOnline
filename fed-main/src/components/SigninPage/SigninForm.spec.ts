import { render, cleanup, fireEvent } from '@testing-library/vue'
import { describe, it, expect, beforeAll, vi } from 'vitest'
import { createRouter, createWebHistory, Router } from 'vue-router'
import { shallowMount } from '@vue/test-utils'
import SigninForm from './SigninForm.vue'
import { initializeFirebaseApp } from '../../config/firebase.config'
import { Auth, getAuth } from 'firebase/auth'
import { pinia } from '../../main'
import { TestUtils } from '../../utils/helpers'
import { useValidator } from '../../utils/composables/validator'
import { default as useStatePhoneVerification } from '../PhoneVerificationPage/useState'
import { useAuthStore } from '../../store/auth'
import { ComponentPublicInstance } from 'vue'

let router: Router
let auth: Auth

// Set the return of a function when we sign in user with phone
const signInUserWithPhoneMockData = {
  uid: 'ZU4jn14tbWfspKuWorshlqQ1s2E3',
  firstName: 'Foo',
  lastName: 'Bar',
  role: 'USER_DEFAULT',
  email: null,
  phoneNumber: null,
  displayName: 'Foo Bar',
  photoURL: null,
  emailVerified: false,
}

describe('SigninForm', async () => {
  beforeAll(() => {
    router = createRouter({
      scrollBehavior(to, from, savedPosition) {
        if (savedPosition) {
          return savedPosition
        } else {
          return { top: 0 }
        }
      },
      history: createWebHistory(),
      routes: [
        {
          path: '/',
          name: 'home',
          components: {},
        },
        {
          path: '/email-verification',
          name: 'email-verification',
          components: {},
        },
        {
          path: '/phone-verification',
          name: 'phone-verification',
          components: {},
        },
      ],
    })
    auth = getAuth(initializeFirebaseApp(process.env as ImportMetaEnv))
  })

  it('render component correctly and signin a user with phone, email or google', async () => {
    const wrapper = render(SigninForm, {
      global: {
        plugins: [pinia, router],
        provide: {
          $auth: auth,
        },
      },
    })

    const buttonSignInAUser = wrapper.getByTestId('Sign in a user')
    const buttonSigninWithGoogle = wrapper.getByTestId('Sign in with Google')

    // Mock signUpWithGoogle() function
    TestUtils.Vitest.mockImplementationClick(buttonSigninWithGoogle, () => undefined)

    // Mock signInUser() function
    TestUtils.Vitest.mockImplementationSubmit(buttonSignInAUser, () => undefined)

    await fireEvent.submit(buttonSignInAUser)
    await fireEvent.click(buttonSigninWithGoogle)

    expect(buttonSignInAUser).toBeTruthy()
    expect(buttonSignInAUser).respondTo('submit')

    expect(buttonSigninWithGoogle).toBeTruthy()
    expect(buttonSigninWithGoogle).respondTo('click')

    cleanup()
  })

  it('call signInUser() function', async () => {
    const wrapper = shallowMount(SigninForm, {
      global: {
        plugins: [pinia, router],
        provide: {
          $auth: auth,
        },
      },
    })

    const wrapperVm = wrapper.vm as ComponentPublicInstance & {
      signInUser: () => void
      signInWithEmail: () => void
      clearInputs: () => void
      phoneOrEmail: string
      signInWithPhone: () => object
    }

    // Mock and test signInUser() and test function
    const spyOnSignInUser = vi.spyOn(wrapperVm, 'signInUser')
    spyOnSignInUser.mockImplementation(() => undefined)
    await wrapperVm.signInUser()

    expect(spyOnSignInUser).toHaveBeenCalled()

    // Test useValidator() function
    const email = 'test@test.com'
    const incorrectEmail = 'test'
    const phone = '+61411111111'

    const testEmail = useValidator({
      phoneOrEmail: email,
    })

    expect(testEmail.validProps.mobilePhone.valid).toBe(false)
    expect(testEmail.validProps.mobilePhone.message).toBe(
      'Please enter a valid Mobile Phone. The phone number must start with a + and must be between 7 and 16 digits.'
    )
    expect(testEmail.validProps.email.valid).toBe(true)
    expect(testEmail.validProps.email.message).toBe('')
    expect(testEmail.validProps.phoneOrEmail.valid).toBe(true)
    expect(testEmail.validProps.phoneOrEmail.message).toBe('')

    const testPhone = useValidator({
      phoneOrEmail: phone,
    })

    expect(testPhone.validProps.mobilePhone.valid).toBe(true)
    expect(testPhone.validProps.mobilePhone.message).toBe('')
    expect(testPhone.validProps.phoneOrEmail.valid).toBe(true)
    expect(testPhone.validProps.phoneOrEmail.message).toBe('')

    const testWithIncorrectEmail = useValidator({
      phoneOrEmail: incorrectEmail,
    })

    expect(testWithIncorrectEmail.validProps.mobilePhone.valid).toBe(false)
    expect(testWithIncorrectEmail.validProps.mobilePhone.message).toBe(
      'Please enter a valid Mobile Phone. The phone number must start with a + and must be between 7 and 16 digits.'
    )
    expect(testWithIncorrectEmail.validProps.email.valid).toBe(false)
    expect(testWithIncorrectEmail.validProps.email.message).toBe('Please enter a valid Email.')
    expect(testWithIncorrectEmail.validProps.phoneOrEmail.valid).toBe(false)
    expect(testWithIncorrectEmail.validProps.phoneOrEmail.message).toBe(
      'Please enter a valid Mobile Number or Email. The phone number must start with a + and must be between 7 and 16 digits.'
    )

    // Mock and test signInWithEmail() and test function
    const spyOnSignInWithEmail = vi.spyOn(wrapperVm, 'signInWithEmail')
    spyOnSignInWithEmail.mockImplementation(() => undefined)
    await wrapperVm.signInWithEmail()

    expect(spyOnSignInWithEmail).toHaveBeenCalled()

    // Test clearInputs() function
    const spyOnClearInputs = vi.spyOn(wrapperVm, 'clearInputs')

    await wrapper.find('[data-testid="Phone or Email"]').setValue('+61411111111')

    expect(wrapperVm.phoneOrEmail).toBe('+61411111111')

    await wrapperVm.clearInputs()

    expect(wrapperVm.phoneOrEmail).toBe('')
    expect(spyOnClearInputs).toHaveBeenCalled()

    // Mock and test signInWithPhone() and test function
    const spyOnSignInWithPhone = vi.spyOn(wrapperVm, 'signInWithPhone')
    spyOnSignInWithPhone.mockImplementation(() => signInUserWithPhoneMockData)
    await wrapperVm.signInWithPhone()

    expect(spyOnSignInWithPhone).toHaveBeenCalled()
    expect(spyOnSignInWithPhone).toReturnWith(signInUserWithPhoneMockData)

    // Test setting phone verification useStore()
    await wrapper.find('[data-testid="Phone or Email"]').setValue('+61411111111')

    const { fullName: name, phoneNumber, setFullName, setPhoneNumber } = useStatePhoneVerification()
    setFullName(signInUserWithPhoneMockData.displayName)
    setPhoneNumber(wrapperVm.phoneOrEmail)

    expect(name.value).toBe('Foo Bar')
    expect(phoneNumber.value).toBe('+61411111111')

    spyOnSignInUser.mockReset()
    spyOnSignInWithEmail.mockReset()
    spyOnSignInWithPhone.mockReset()
    spyOnClearInputs.mockReset()
  })

  it('call signUpWithGoogle() function', async () => {
    const wrapper = shallowMount(SigninForm, {
      global: {
        plugins: [pinia, router],
        provide: {
          $auth: auth,
        },
      },
    })

    const wrapperVm = wrapper.vm as ComponentPublicInstance & {
      signUpWithGoogle: () => void
      storeUserToDatabase: () => void
    }

    // Mock and test signUpWithGoogle() and test function
    const spyOnSignUpWithGoogle = vi.spyOn(wrapperVm, 'signUpWithGoogle')
    spyOnSignUpWithGoogle.mockImplementation(() => undefined)
    await wrapperVm.signUpWithGoogle()

    expect(spyOnSignUpWithGoogle).toHaveBeenCalled()

    // Mock and test storeUserToDatabase() and test function
    const spyOnStoreUserToDatabase = vi.spyOn(wrapperVm, 'storeUserToDatabase')
    // Set the auth store with pinia
    const useStoreAuth = useAuthStore(pinia)
    spyOnStoreUserToDatabase.mockImplementation(() => {
      useStoreAuth.user = {
        uid: 'ZU4jn14tbWfspKuWorshlqQ1s2E3',
        firstName: 'Foo',
        lastName: 'Bar',
        role: 'USER_DEFAULT',
        email: 'test@test.com',
        phoneNumber: null,
        displayName: 'Foo Bar',
        photoURL: null,
        emailVerified: true,
      }
    })
    await wrapperVm.storeUserToDatabase()

    expect(spyOnStoreUserToDatabase).toHaveBeenCalled()
    expect(useStoreAuth.user).toEqual({
      uid: 'ZU4jn14tbWfspKuWorshlqQ1s2E3',
      firstName: 'Foo',
      lastName: 'Bar',
      role: 'USER_DEFAULT',
      email: 'test@test.com',
      phoneNumber: null,
      displayName: 'Foo Bar',
      photoURL: null,
      emailVerified: true,
    })
  })

  it('snap shot matches', async () => {
    const wrapper = render(SigninForm, {
      global: {
        plugins: [pinia, router],
        provide: {
          $auth: auth,
        },
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
