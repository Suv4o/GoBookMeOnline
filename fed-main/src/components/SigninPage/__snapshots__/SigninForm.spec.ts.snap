// Vitest Snapshot v1

exports[`SigninForm > snap shot matches 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="mx-auto w-full max-w-sm lg:w-96"
      >
        <h2
          class="mt-6 text-3xl font-extrabold text-gray-900"
        >
          Sign in to your account
        </h2>
        <div
          class="mt-8"
        >
          <div>
            <div>
              <p
                class="text-sm font-medium text-gray-700"
              >
                Sign in with Google
              </p>
              <div
                class="mt-1 grid grid-cols-1 gap-3"
              >
                <div>
                  <a
                    class="w-full inline-flex justify-center py-2 px-4 border border-gray-300 rounded-md shadow-sm bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"
                    data-testid="Sign in with Google"
                    href="javascript:;"
                  >
                    <span
                      class="sr-only"
                    >
                      Sign in with Google
                    </span>
                    <svg
                      aria-hidden="true"
                      class="w-5 h-5"
                      fill="currentColor"
                      viewBox="0 0 210 210"
                    >
                      <path
                        d="M0,105C0,47.103,47.103,0,105,0c23.383,0,45.515,7.523,64.004,21.756l-24.4,31.696C133.172,44.652,119.477,40,105,40
	c-35.841,0-65,29.159-65,65s29.159,65,65,65c28.867,0,53.398-18.913,61.852-45H105V85h105v20c0,57.897-47.103,105-105,105
	S0,162.897,0,105z"
                      />
                    </svg>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="mt-6 relative"
            >
              <div
                aria-hidden="true"
                class="absolute inset-0 flex items-center"
              >
                <div
                  class="w-full border-t border-gray-300"
                />
              </div>
              <div
                class="relative flex justify-center text-sm"
              >
                <span
                  class="px-2 bg-gray-50 text-gray-500"
                >
                   Or continue with 
                </span>
              </div>
            </div>
          </div>
          <div
            class="mt-6"
          >
            <form
              class="space-y-6"
              data-testid="Sign in a user"
            >
              <div>
                <label
                  class="block text-sm font-medium text-gray-700"
                  for="mobile-or-email"
                >
                   Mobile Number or Email 
                </label>
                <div
                  class="mt-1"
                >
                  <input
                    autocomplete="email|tel"
                    class="appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none sm:text-sm focus:ring-teal-500 focus:border-teal-500"
                    data-testid="Phone or Email"
                    id="mobile-or-email"
                    name="mobile-or-email"
                    required=""
                    type="text"
                  />
                  <!--v-if-->
                </div>
              </div>
              <div>
                <button
                  class="w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-white bg-teal-600 hover:bg-teal-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
                  type="submit"
                >
                  <!--v-if-->
                   Sign in
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto w-full max-w-sm lg:w-96"
    >
      <h2
        class="mt-6 text-3xl font-extrabold text-gray-900"
      >
        Sign in to your account
      </h2>
      <div
        class="mt-8"
      >
        <div>
          <div>
            <p
              class="text-sm font-medium text-gray-700"
            >
              Sign in with Google
            </p>
            <div
              class="mt-1 grid grid-cols-1 gap-3"
            >
              <div>
                <a
                  class="w-full inline-flex justify-center py-2 px-4 border border-gray-300 rounded-md shadow-sm bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"
                  data-testid="Sign in with Google"
                  href="javascript:;"
                >
                  <span
                    class="sr-only"
                  >
                    Sign in with Google
                  </span>
                  <svg
                    aria-hidden="true"
                    class="w-5 h-5"
                    fill="currentColor"
                    viewBox="0 0 210 210"
                  >
                    <path
                      d="M0,105C0,47.103,47.103,0,105,0c23.383,0,45.515,7.523,64.004,21.756l-24.4,31.696C133.172,44.652,119.477,40,105,40
	c-35.841,0-65,29.159-65,65s29.159,65,65,65c28.867,0,53.398-18.913,61.852-45H105V85h105v20c0,57.897-47.103,105-105,105
	S0,162.897,0,105z"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
          <div
            class="mt-6 relative"
          >
            <div
              aria-hidden="true"
              class="absolute inset-0 flex items-center"
            >
              <div
                class="w-full border-t border-gray-300"
              />
            </div>
            <div
              class="relative flex justify-center text-sm"
            >
              <span
                class="px-2 bg-gray-50 text-gray-500"
              >
                 Or continue with 
              </span>
            </div>
          </div>
        </div>
        <div
          class="mt-6"
        >
          <form
            class="space-y-6"
            data-testid="Sign in a user"
          >
            <div>
              <label
                class="block text-sm font-medium text-gray-700"
                for="mobile-or-email"
              >
                 Mobile Number or Email 
              </label>
              <div
                class="mt-1"
              >
                <input
                  autocomplete="email|tel"
                  class="appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none sm:text-sm focus:ring-teal-500 focus:border-teal-500"
                  data-testid="Phone or Email"
                  id="mobile-or-email"
                  name="mobile-or-email"
                  required=""
                  type="text"
                />
                <!--v-if-->
              </div>
            </div>
            <div>
              <button
                class="w-full flex justify-center py-2 px-4 border border-transparent rounded-md shadow-sm text-sm font-medium text-white bg-teal-600 hover:bg-teal-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
                type="submit"
              >
                <!--v-if-->
                 Sign in
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
